<launch>

<!-- name: ['left_e0', 'left_e1', 'left_s0', 'left_s1', 'left_w0', 'left_w1', 'left_w2']
position: [-0.44830588525951215, 1.8465293734170667, -0.004601942363656241, -1.0043739208679747, 0.34476218207724674, 0.8018884568671001, -3.0418839023767754] -->

  <!-- Good position for the arm so the camera can see all 3 objects -->
  <!-- <param name="left_w0" value="0.34476218207724674"/>
  <param name="left_w1" value="0.8018884568671001"/>
  <param name="left_w2" value="-3.0418839023767754"/>
  <param name="left_e0" value="-0.44830588525951215"/>
  <param name="left_e1" value="1.8465293734170667"/>
  <param name="left_s0" value="-0.004601942363656241"/>
  <param name="left_s1" value="-1.0043739208679747"/> -->

  <!-- Better config for RPL Lab -->
  <param name="right_e0" value="0.6787864986392956"/>
  <param name="right_e1" value="1.5013836961428486"/>
  <param name="right_s0" value="0.23930100291012454"/>
  <param name="right_s1" value="-0.9997719785043184"/>
  <param name="right_w0" value="-0.40382044241083515"/>
  <param name="right_w1" value="1.175029283520227"/>
  <param name="right_w2" value="-2.9582819494370205"/>

  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="1.28 0.10 0.19 0 0 0 base camera_link 1000" />

  <include file="$(find baxter_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <include file="$(find baxter_moveit_config)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
  </include>

  <!-- Specifies which camera to use and the resolution -->
  <node name="camera_control" pkg="baxter_tools" type="camera_control.py"
    args="-o right_hand_camera -r 640x400"/>

  <!-- Gripper Argument -->
  <!-- might need to change this to left -->
  <arg name="gripper" default="right" />

  <!-- Start the Gripper Action Server -->
  <node name="rsdk_gripper_action_server" pkg="baxter_interface" type="gripper_action_server.py" args="--gripper $(arg gripper)" required="true" />

  <node pkg="baxter_interface" type="joint_trajectory_action_server.py" name="joint_traj_action_srv" output="screen" respawn="false" />

  <!-- up to date -->
  <node pkg="assembly_task" type="baxter_object_prediction.py" name="baxter_object_prediction" output="screen" respawn="true"/>

  <!-- need to modify this somehow -->
  <!-- <node pkg="object_recognition" type="pickup_object.py" name="pickup_object" output="screen" respawn="true" launch-prefix="xterm -e "/> -->
  <node pkg="assembly_task" type="pickup_object.py" name="pickup_object" output="screen" respawn="false" />

  <node pkg="assembly_task" type="request_object.py" name="request_object" output="screen" respawn="false" required="true"/>

  <!-- Gripper Cuff control -->
  <node pkg="baxter_examples" type="gripper_cuff_control.py" name="gripper_cuff_control" output="screen" respawn="false" />

</launch>
